siliconcompiler/core.py

```diff
@@ -559,6 +577,7 @@ def create_cmdline(self,
                 args = [free_keys.pop(0) if key == 'default' else key for key in keypath]
 
                 # Remainder is the value we want to set, possibly with a step/index value beforehand
+                sctype = self.get(*keypath, field='type')
                 pernode = self.get(*keypath, field='pernode')
                 step, index = None, None
                 if pernode == 'required':
```

Comment:
This one is also unexpected to me.
I also couldn't find the possible values of `pernode` in the documentation.

Reply:
looks like that is missing the docs, I'll add it